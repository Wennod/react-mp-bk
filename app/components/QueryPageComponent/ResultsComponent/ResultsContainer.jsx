import React from 'react';

import ResultList from './ResultsListComponent/ResultList';
import ResultsFilers from './ResultsFiltersComponent/ResultsFilters';

class ResultsContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="results">
                <ResultsFilers isData={this.props.data && this.props.data.length ? this.props.data.length : false} />
                <ResultList data={this.props.data} />
            </div>
        );
    }
}

export default ResultsContainer;