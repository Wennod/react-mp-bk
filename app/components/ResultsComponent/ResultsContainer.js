import React from 'react';

class ResultsContainer extends React.Component {
    render() {
        
    }
}

export default ResultsContainer;