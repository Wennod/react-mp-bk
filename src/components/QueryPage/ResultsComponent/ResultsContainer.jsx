import React from 'react';

import ResultList from './ResultsListComponent/ResultList';
import ResultsFilters from './ResultsFiltersComponent/ResultsFilters';

import Style from './ResultsContainer.style.scss';

class ResultsContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className={this.props.data && this.props.data.length ? 'results' : 'results results--nodata'}>
                <div className="results__filter-container">
                    <ResultsFilters isData={this.props.data && this.props.data.length ? this.props.data.length : false} />
                </div>
                <div className="results__list-container">
                    <ResultList data={this.props.data} />
                </div>
            </div>
        );
    }
}

export default ResultsContainer;