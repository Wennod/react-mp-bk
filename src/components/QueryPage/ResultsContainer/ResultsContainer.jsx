import React from 'react';
import { connect } from 'react-redux';

import ResultList from './ResultsListComponent/ResultList';
import ResultsFilters from './ResultsFiltersComponent/ResultsFilters';

import Style from './ResultsContainer.style.scss';

class ResultsContainer extends React.Component {

    render() {
        let results = this.props.results.queryResults.data;

        return (
            <div 
                className={results && results.length 
                    ? 'results' 
                    : 'results results--nodata'
                }
            >
                <div className="results__filter-container">
                    <ResultsFilters 
                        isData={results && results.length 
                            ? results.length 
                            : false} 
                    />
                </div>
                <div className="results__list-container">
                    <ResultList data={results} />
                </div>
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        results: state.query
    }
}

export default connect(mapStateToProps)(ResultsContainer);
