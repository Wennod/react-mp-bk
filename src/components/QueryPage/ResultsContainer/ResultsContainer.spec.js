import React from 'react';
import ResultsContainer from './ResultsContainer';

const __data = [
    {
        "id": 412302,
        "title": "Gemini",
        "tagline": "",
        "vote_average": 10,
        "vote_count": 3,
        "release_date": "2018-03-30",
        "poster_path": "https: //image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg",
        "budget": 0,
        "revenue": 0,
        "genres":  ["Mystery","Thriller"],
        "runtime": 92
    }
]

describe('ResultsContainer snapshot', () => {
    it('should match snapshot', () => {
        const tree = mount(<ResultsContainer />);
        expect(tree).toMatchSnapshot();
    });
});

describe('ResultsContainer', () => {
    it('should receive props', () => {
        const tree = mount(<ResultsContainer data={__data} />);
        expect(tree.props().data && tree.props().data.length).toBeTruthy();
    });

    it('should behave properly in case if no data received', () => {
        const tree = mount(<ResultsContainer data={__data} />);
        expect(tree.props().data && tree.props().data.length).toBeFalsy;
    });
});