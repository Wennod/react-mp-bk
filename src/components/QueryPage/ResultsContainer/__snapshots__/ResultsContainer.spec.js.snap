// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsContainer snapshot should match snapshot 1`] = `
<ResultsContainer>
  <div
    className="results results--nodata"
  >
    <div
      className="results__filter-container"
    >
      <ResultsPane
        isData={false}
      >
        <div
          className="result-filter result-filter--nodata"
        />
      </ResultsPane>
    </div>
    <div
      className="results__list-container"
    >
      <ResultList>
        <div
          className="list"
        >
          <h2
            className="results__error"
          >
            No results
          </h2>
        </div>
      </ResultList>
    </div>
  </div>
</ResultsContainer>
`;
